packages/sdk/src/modules/indexing/indexer.ts

```diff
@@ -0,0 +1,77 @@
+import axios from 'axios';
+import fs from 'fs';
+import { ApillonModel } from '../../lib/apillon';
+import { ApillonApi } from '../../lib/apillon-api';
+import { compressIndexerSourceCode } from '../../util/indexer-utils';
+import { IDeployIndexer } from '../../types/indexer';
+
+export class Indexer extends ApillonModel {
+  /**
+   * User assigned name of the indexer.
+   */
+  public name: string = null;
+
+  /**
+   * User assigned description of the indexer.
+   */
+  public description: string = null;
+
+  /**
+   * Constructor which should only be called via Indexing class.
+   * @param uuid Unique identifier of the indexer.
+   * @param data Data to populate the indexer with.
+   */
+  constructor(uuid: string, data?: Partial<Indexer>) {
+    super(uuid);
+    this.API_PREFIX = `/indexing/indexers/${uuid}`;
+    this.populate(data);
+  }
+
+  /**
+   * Prepare indexer source code, upload it to s3 and deploy the indexer.
+   * @param path Path to the indexer source code directory.
+   */
+  public async deployIndexer(path: string): Promise<any> {
+    //Check directory and if squid.yaml exists in it
+    if (!fs.existsSync(path)) {
+      throw new Error('Path does not exist');
```

Comment:
Better to just use `return console.error(...)`, no need to throw since that would just fill up the terminal logs with stack trace which is not necessary.